src/grt/src/GlobalRouter.cpp

```diff
@@ -1610,13 +1610,28 @@ void GlobalRouter::applyObstructionAdjustment(const odb::Rect& obstruction,
                                          false,
                                          tech_layer->getDirection());
 
+  int grid_limit = vertical ? grid_->getYGrids() : grid_->getXGrids();
   if (!vertical) {
+    // if obstruction is inside a single gcell, block the edge between current
+    // gcell and the adjacent gcell
+    if (first_tile.getX() == last_tile.getX()
+        && last_tile.getX() + 1 < grid_limit) {
+      int last_tile_x = last_tile.getX() + 1;
+      last_tile.setX(last_tile_x);
+    }
```

Comment:
I think generally not just for first_tile.getX() == last_tile.getX():
```
last_tile.setX(last_tile.getX() + 1);
first_tile.setX(first_tile.getX() - 1);
```

Reply:
I've tried this approach, but it is too agressive. A bunch of designs failed with high congestion. We are probably blocking pin access for macro pins and instance pins that are above layer 1.

My understanding is that blocking to the right/top is enought because the way FastRoute algorithm works, creating the edges in these two directions. Also, the test case used to validate this update is showing the correct congestion in layer 1 and we don't have guides in this layer anymore.